src/ui/verificationSymbolStatusView.ts

```diff
@@ -0,0 +1,318 @@
+      const items: TestItem[] = this.getItemsInRun(request, controller);
+      const runningItems: TestItem[] = [];
+      let outerResolve: () => void;
+      await this.processUpdateLock;
+      this.processUpdateLock = new Promise((resolve) => {
+        outerResolve = resolve;
+      });
+
+      const runs = items.map(item => this.languageClient.runVerification({ position: item.range!.start, textDocument: { uri: item.uri!.toString() } }));
+      for(const index in runs) {
+        const success = await runs[index];
+        if(success) {
+          runningItems.push(items[index]);
+        }
+      }
+
+      if(runningItems.length > 0) {
+        this.createRun(runningItems);
+      }
+      outerResolve!();
+    }, true);
+    return controller;
+  }
+
+  private getItemsInRun(run: TestRunRequest, controller: TestController): TestItem[] {
+    const allItems: TestItem[] = [];
+    controller.items.forEach(item => allItems.push(item));
+    const result: TestItem[] = [];
+    const todo = run.include ? [ ...run.include ] : allItems;
+    const excludes = run.exclude ? new Set(run.exclude) : new Set<TestItem>();
+    while(todo.length > 0) {
+      const current = todo.pop()!;
+      if(excludes.has(current)) {
+        continue;
+      }
+
+      current.children.forEach(child => {
+        todo.push(child);
+      });
+      if(current.children.size === 0) {
+        result.push(current);
+      }
+    }
+    return result;
+  }
+
+  private createRun(items: TestItem[]): TestRun {
+    const run = this.controller.createTestRun(new TestRunRequest(items));
+    for(const item of items) {
+      this.itemRuns.set(item.id, { run });
+    }
+    run.token.onCancellationRequested(() => {
+      if(!this.automaticRunEnd) {
+        for(const item of items) {
+          this.languageClient.cancelVerification({ position: item.range!.start, textDocument: { uri: item.uri!.toString() } });
+        }
+      }
+    });
+    this.runItemsLeft.set(run, items.length);
+    return run;
+  }
+
+  private async update(params: IVerificationSymbolStatusParams): Promise<void> {
+    await this.processUpdateLock;
+
+    this.updateStatusBar(params);
+    this.getFirstStatusForFilePromise(params.uri).resolve(params);
+    this.updatesPerFile.set(params.uri, params);
+    const uri = Uri.parse(params.uri);
+    const controller = this.controller;
+
+    const rootSymbols = await commands.executeCommand('vscode.executeDocumentSymbolProvider', uri) as DocumentSymbol[];
+    let items: TestItem[];
+    const document = await workspace.openTextDocument(uri.fsPath);
+    if(rootSymbols !== undefined) {
+
+      items = this.updateUsingSymbols(params, document, controller, rootSymbols);
+    } else {
+      items = params.namedVerifiables.map(f => this.getItem(document,
+        VerificationSymbolStatusView.convertRange(f.nameRange), controller, uri));
+    }
+
+    const runningItemsWithoutRun = params.namedVerifiables.
```

Comment:
What do you mean here by "runningItemsWithoutRun"? Can you make it more explicit or add a comment?